### Урок 4. Непрерывные случайные величины. Функция распределения и плотность распределения вероятностей. Равномерное и нормальное распределение. Центральная предельная теорема

#### Задача 1
О непрерывной равномерно распределенной случайной величине B известно, что ее дисперсия равна 0.2. Можно ли найти правую границу величины B и ее среднее значение зная, что левая граница равна 0.5? Если да, найдите ее.

#### Решение 1

$$D(X) = \frac{(b - a)^2}{12}$$

D(X) = 0.2

a = 0.5

b = ?

mean = ?

In [4]:
from math import sqrt

a = 0.5
dX = 0.2
b = sqrt(12 * dX) + a
b

2.049193338482967

Правая граница b = 2.049193338482967

In [7]:
x_mean = (a + b) / 2
x_mean

1.2745966692414834

среднее значение = 1.2745966692414834

#### Задача 2
Рост взрослого населения города X имеет нормальное распределение, причем, средний рост равен 174 см, а среднее квадратическое отклонение равно 8 см. посчитайте, какова вероятность того, что случайным образом выбранный взрослый человек имеет рост:
1. больше 182 см?
2. больше 190 см?
3. от 166 см до 190 см?
4. от 166 см до 182 см?
5. от 158 см до 190 см?
6. не выше 150 см или не ниже 190 см?
7. не выше 150 см или не ниже 198 см?
8. ниже 166 см?
Задачу можно решить двумя способами: без использования сторонних библиотек (numpy, scipy, pandas и пр.), а затем проверить себя с помощью встроенных функций

#### Решение 2

In [48]:
loc = 174
scale = 8

import numpy as np
from math import erf, pow
from scipy import stats

def cdf_normal(x):
    return (1 + erf((x - loc) / sqrt(2*pow(scale, 2)))) / 2

def more_than(a):
    result = 1 - cdf_normal(a)
    result_stats = 1 - stats.norm.cdf(a, loc, scale)
    print(f'{result} / {result == result_stats}')
    
def less_than(a):
    result = cdf_normal(a)
    result_stats = stats.norm.cdf(a, loc, scale)
    print(f'{result} / {result == result_stats}')
    
def in_range(a, b):
    result = cdf_normal(b) - cdf_normal(a)
    result_stats = stats.norm.cdf(b, loc, scale) - stats.norm.cdf(a, loc, scale)
    print(f'{result} / {result == result_stats}')
    
def not_in_range(a, b):
    result = 1 - (cdf_normal(b) - cdf_normal(a))
    result_stats = 1 - (stats.norm.cdf(b, loc, scale) - stats.norm.cdf(a, loc, scale))
    print(f'{result} / {result == result_stats}')

In [37]:
# больше 182 см
more_than(182)

0.15865525393145707 / True


In [38]:
# больше 190 см
more_than(190)

0.02275013194817921 / True


In [40]:
# от 166 см до 190 см
in_range(166, 190)

0.8185946141203637 / True


In [41]:
# от 166 см до 182 см
in_range(166, 182)

0.6826894921370859 / True


In [42]:
# от 158 см до 190 см
in_range(158, 190)

0.9544997361036416 / True


In [45]:
# не выше 150 см или не ниже 190 см
not_in_range(150, 190)

0.024100029979809312 / True


In [46]:
# не выше 150 см или не ниже 198 см
not_in_range(150, 198)

0.002699796063260207 / True


In [49]:
# ниже 166 см
less_than(166)

0.15865525393145707 / True
